### DISCUSSION TITLE
How to specify the IconName in BarIcon dynamically?

### DISCUSSION THREAD

**Discussion by bbuchanan** [Megabit/Blazorise#1954] [Q&A]
I hope this is just my lack of experience in Blazor syntax, but I'm using the BarIcon and everything works great when I do this:

```cshtml
<BarItem>
  <BarLink To=\"\">
    <BarIcon IconName="FontAwesomeIcons.Database" />
    My Page
  </BarLink>
</BarItem>
```

But I'm attempting to define routes dynamically. If I do this same where route.IconName is the string FontAwesomeIcons.Database, nothing shows up. Since FontAwesomeIcons is a helper class that returns a string, how can I use it in a dynamic sense.

```cshtml
@foreach (var route in routes)
{
  <BarItem>
    <BarLink To=\"@route.Route\">
      <BarIcon IconName="@route.IconName" />
      @route.RouteName
    </BarLink>
  </BarItem>
}
```

**Comment by stsrki** [comment]
Might be a problem with Blazor and for loop. You can use temp variables.

```cshtml
@foreach (var route in routes)
{
  var iconName = route.IconName;
  var to = route.Route;
  var routeName = route.RouteName;

  <BarItem>
    <BarLink To="@to">
      <BarIcon IconName="@iconName " />
      @routeName
    </BarLink>
  </BarItem>
}
```

**Reply by bbuchanan** [reply]
All of other variables work without a temp variable. When I assign the iconName to a temp variable as you show, it still doesn't work. I'm going to just use the underlying string value for now.

**Reply by stsrki** [reply]
Not sure what can be wrong. I've tried with `<BarIcon IconName="@("fa-database")" />` and it works just fine.

What Blazorise version you have?
